pkg/component/kubernetes/apiserverexposure/sni.go

```diff
@@ -142,17 +200,46 @@ func (s *sni) Deploy(ctx context.Context) error {
 		if err := managedresources.CreateForSeed(ctx, s.client, s.namespace, managedResourceName, false, serializedObjects); err != nil {
 			return err
 		}
+	} else {
+		if err := managedresources.DeleteForSeed(ctx, s.client, s.namespace, managedResourceName); err != nil {
+			return err
+		}
+	}
+
+	var destinationMutateFn func() error
+	destinationMutateFn = istio.DestinationRuleWithLocalityPreference(destinationRule, getLabels(), hostName)
+	if features.DefaultFeatureGate.Enabled(features.IstioTLSTermination) {
+		destinationMutateFn = istio.DestinationRuleWithTLSTermination(destinationRule, getLabels(), hostName, s.valuesFunc().Hosts[0], s.namespace+apiserver.IstioCASecretSuffix, istioapinetworkingv1beta1.ClientTLSSettings_SIMPLE)
 	}
 
-	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, s.client, destinationRule, istio.DestinationRuleWithLocalityPreference(destinationRule, getLabels(), hostName)); err != nil {
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, s.client, destinationRule, destinationMutateFn); err != nil {
 		return err
 	}
 
-	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, s.client, gateway, istio.GatewayWithTLSPassthrough(gateway, getLabels(), s.valuesFunc().IstioIngressGateway.Labels, s.valuesFunc().Hosts, kubeapiserverconstants.Port)); err != nil {
+	if features.DefaultFeatureGate.Enabled(features.IstioTLSTermination) {
+		destinationMTLSMutateFn := istio.DestinationRuleWithTLSTermination(mTLSDestinationRule, getLabels(), mTLSHostName, s.valuesFunc().Hosts[0], s.namespace+apiserver.IstioCASecretSuffix, istioapinetworkingv1beta1.ClientTLSSettings_MUTUAL)
```

Comment:
Same question about the host as above.

Reply:
see https://github.com/gardener/gardener/pull/11085#discussion_r1920379798